benchmarks/spiegelman_et_al_2016_gcubed/spiegelman_material.cc

```diff
@@ -0,0 +1,680 @@
+        cohesion = get_vector_double("List of cohesions",n_fields,prm);
+        phi = get_vector_double("List of angles of internal friction",n_fields,prm);
+
+        sin_phi.resize(n_fields);
+        cos_phi.resize(n_fields);
+        for (unsigned int c = 0; c < n_fields; ++c)
+          {
+            sin_phi[c] = std::sin(phi[c] * numbers::PI/180);
+            cos_phi[c] = std::cos(phi[c] * numbers::PI/180);
+          }
+
+        prefactor = get_vector_double("List of prefactors",n_fields,prm);
+        stress_exponent = get_vector_double("List of stress exponents",n_fields,prm);
+      }
+      prm.leave_subsection();
+
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection ("Drucker prager compositions");
+        {
+
+          // Reference and minimum/maximum values
+          reference_T = prm.get_double("Reference temperature");
+          ref_visc = prm.get_double ("Reference viscosity");
+          min_strain_rate = get_vector_double("Minimum strain rate",n_fields,prm);
+          min_visc = get_vector_double ("Minimum viscosity",n_fields,prm);
+          max_visc = get_vector_double ("Maximum viscosity",n_fields,prm);
+          veff_coefficient = get_vector_double ("Effective viscosity coefficient",n_fields,prm);
+          reference_compressibility  = prm.get_double ("Reference compressibility");
+          use_deviator_of_strain_rate = prm.get_bool ("Use deviator of strain-rate");
+
+
+          // averaging parameters
+          viscosity_averaging_p = prm.get_double("Viscosity averaging p");
+
+          use_analytical_derivative = prm.get_bool("Use analytical derivative");
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+
+      // Declare dependencies on solution variables
+      this->model_dependence.viscosity = NonlinearDependence::temperature | NonlinearDependence::pressure | NonlinearDependence::strain_rate | NonlinearDependence::compositional_fields;
+      this->model_dependence.density = NonlinearDependence::temperature | NonlinearDependence::pressure | NonlinearDependence::compositional_fields;
+      this->model_dependence.compressibility = NonlinearDependence::none;
+      this->model_dependence.specific_heat = NonlinearDependence::none;
+      this->model_dependence.thermal_conductivity = NonlinearDependence::temperature | NonlinearDependence::pressure | NonlinearDependence::compositional_fields;
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    ASPECT_REGISTER_MATERIAL_MODEL(DruckerPragerCompositions,
+                                   "drucker prager compositions",
+                                   " An implementation of the spiegelman 2016 benchmark paper. It "
+                                   "implements a regularized Drucker Prager material and a linear "
+                                   "material. The derivatives for the Newton solver have been "
```

Comment:
what does "linear material" reference here? Is it a DP material or a linear material?

Reply:
The Spiegelman benchmark requires a combination of both. I clarified it a bit.